### ISSUE TITLE
Using secrets with _export

### ISSUE THREAD

**Issue by nickhall** [treasure-data/digdag#926]
I'm using digdag with embulk to upload data to Treasure Data, and I'd like to set a few credential-related variables in the root of the workflow to share across all of the embulk tasks. Currently I can reference the secrets directly in each task and set the environment variables in the command, like`sh>: var=${secret:my_var} embulk run file.yml`, but when I'm syncing dozens of tables to TD, I'd like to just set the environment variables from within digdag and reference them within the embulk file. Using normal parameters this is possible, but not secrets.

This is what I want to accomplish:

```
_export:
  TD_API_KEY: ${secret:td_api_key}

+data_upload:
  +task_1:
    sh>: embulk run my_file.yml
  +task_2:
    sh> embulk run my_other_file.yml
```

However, `TD_API_KEY` is assigned the literal value `${secret:td_api_key}` and is not expanded. Is this intentional? Or is there a better way to do it?

**Comment by hiroyuki-sato** [comment]
Hello, @nickhall 

The `${}` syntax evaluate task at execution time. And you can't use it in `_export:` section.

Support team in Arm treasure data may propose a better solution for you.
Could you ask it in TD?

**Comment by hadrianhu** [comment]
@nickhall - This is Ivan w/ Treasure Data Support.  A possible solution could be using the liquid syntax to include environment variables in your yml files.  You can read more about it here: https://support.treasuredata.com/hc/en-us/articles/360001502447-Using-Variables-with-Bulk-Import

Please feel free to reach out to us directly at support@treasure-data.com with any questions.

**Comment by nickhall** [comment]
@hadrianhu Thank you for the suggestion. That is what I'm trying to do, but the problem is that I have no way of setting the environment variables from within digdag, which is the main part of what I'm trying to do. The root of the problem is that there is no convenient way to load environment variables without a restart of the system, so I'm trying to use secrets as per the recommendation of TD support, but I can't set secrets to environment variables to use across tasks. I'm at a loss here with what to do.

**Comment by friendofasquid** [comment]
Are you aware of  ‘_env’. This is exactly designed for your use case and should allow you to set secrets. We use this pattern with liquid variables in Embulk to send data to treasure data and works great.

**Comment by doctor** [comment]
Hello,
I tried this in in my workflow. But I am getting an error. Details below.

**Config:**
`+run_etl:
    _env:
      datahub_password: ${secret:datahub_password}
      gmcc_password: ${secret:gmcc_password}

    +truncate_tables:
      py>: tasks.dealer.run_etl
      from_id: 0
      to_id: 0
      flow: 'emb_truncate_tables'`

**Error:**
rror: io.digdag.core.repository.ModelValidationException: Validating workflow failed
+gm_etl+run_etl contains invalid keys: '_env': "{"_env":{"datahub_password":"${secret:datahub_password}","gmcc_password":"${secret:gmcc_password}"}}" (config)

**Comment by doctor** [comment]
Hello, I tried similar way to pass secrets to the subtasks, but it fails.

`
echo "hello"
`

**Comment by doctor** [comment]
Hello @hiroyuki-sato 
I tried the same approach to pass secret, i am getting errors.
```yaml
+run_etl:
  _env:
    datahub_password: ${secret:datahub_password}
    gmcc_password: ${secret:gmcc_password}

  +truncate_tables:
    py>: tasks.dealer.run_etl
    from_id: 0
    to_id: 0
    flow: 'emb_truncate_tables'
```

**Error:** 
```log
error: io.digdag.core.repository.ModelValidationException: Validating workflow failed
+gm_etl+run_etl contains invalid keys: '_env': "{"_env":{"datahub_password":"${secret:datahub_password}","gmcc_password":"${secret:gmcc_password}"}}" (config)
```

**Comment by hiroyuki-sato** [comment]
Hello, @doctor. Have you ever tried the following?


```yaml
  +truncate_tables:
    py>: tasks.dealer.run_etl
    from_id: 0
    to_id: 0
    flow: 'emb_truncate_tables'
    _env:
      datahub_password: ${secret:datahub_password}
      gmcc_password: ${secret:gmcc_password}
```

As I comment before, 

> The `${}` syntax evaluate task at execution time. And you can't use it in `_export`: section.

I suggest you create a new Issue, not comment on the closed issue.

**Comment by doctor** [comment]
Hello @hiroyuki-sato 
Thanks for replying. I did not explain it clearly the issue I am facing. In addition to truncate_tables task, I have 5 more embulk tasks that requires the passwords from secret. Do I need to set the _env: per task to use the secrets or can I set it once in the parent task and reference it in the subtasks?

I also tried creating store params for the passwords, the problem with this approach is that store params are made visible in the UI. In my case passwords are displayed as plain text.

...and yes I will create a new issue going forward. Thanks again.
